File: java/Kafic/src/Pice.java
public class Pice {
    protected String naziv;
    protected String brend;
    protected Ukus ukus;

    public void setUkus(Ukus ukus) {
        this.ukus = ukus;
    }

    public Pice(String naziv, String brend, Ukus ukus) {
        this.naziv = naziv;
        this.brend = brend;
    }

    public String getNaziv() {
        return naziv;
    }

    public void setNaziv(String naziv) {
        this.naziv = naziv;
    }

    public String getBrend() {
        return brend;
    }

    public void setBrend(String brend) {
        this.brend = brend;
    }

    public Ukus getUkus() {
        return ukus;
    }

    public void opisiPice() {
        System.out.println("Pice " + naziv + " | " + brend + ",ukus: " + ukus);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Pice == false) {
            return false;
        }
        Pice p = (Pice) obj;
        if (this.naziv.equals(p.naziv) && this.brend.equals(p.brend) && this.ukus.equals(p.ukus)) {
            return true;
        }
        return false;
    }
}

File: typescript/Kafic/Pice.ts
import { Ukus } from './Ukus';

export class Pice {
  protected naziv: string;
  protected brend: string;
  protected ukus: Ukus;

  public setUkus(ukus: Ukus): void {
      this.ukus = ukus;
  }

  public Pice(naziv: string, brend: string, ukus: Ukus) {
      this.naziv = naziv;
      this.brend = brend;
      this.ukus = ukus;
  }

  public getNaziv(): string {
      return this.naziv;
  }

  public setNaziv(naziv: string): void {
      this.naziv = naziv;
  }

  public getBrend(): string {
      return this.brend;
  }

  public setBrend(brend: string) : void{
      this.brend = brend;
  }

  public getUkus(): Ukus {
      return this.ukus;
  }

  public opisiPice(): void {
      console.log("Pice " + this.naziv + " | " + this.brend + ",ukus: " + this.ukus);
  }

  // @Override
  // public boolean equals(Object obj) {
  //     if (obj instanceof Pice == false) {
  //         return false;
  //     }
  //     Pice p = (Pice) obj;
  //     if (this.naziv.equals(p.naziv) && this.brend.equals(p.brend) && this.ukus.equals(p.ukus)) {
  //         return true;
  //     }
  //     return false;
  // }
}
